### PR TITLE
Range-ify std.string.leftJustify() with leftJustifier()

### PR DESCRIPTION
**Pull request by WalterBright**
http://dlang.org/phobos/std_string.html#.leftJustify

Unit tests at 100%.

### CODE REVIEW

File: std/string.d [hunk 1]
```diff
@@ -2924,39 +2924,129 @@ unittest
...
+{
+    alias C = Unqual!(ElementEncodingType!Range);
+
+    static if (C.sizeof == 1)
+    {
+        import std.utf : byDchar, byChar;
+        return leftJustifier(r.byDchar, width, fillChar).byChar;
     }
-    else
+    else static if (C.sizeof == 2)
```
Comment (JakobOvrum) [inline std/string.d]: Is this path really tested?

**Comment by JakobOvrum** [comment]
LGTM

**Comment by kuettler** [comment]
Do you think consistency with #3224 is worth the trouble?

**Comment by jwhear** [comment]
LGTM

File: std/string.d [hunk 2]
```diff
@@ -2924,39 +2924,129 @@ unittest
     Left justify $(D s) in a field $(D width) characters wide. $(D fillChar)
     is the character that will be used to fill up the space in the field that
     $(D s) doesn't fill.
+
+    Params:
+        s = string
```
Comment (andralex) [inline std/string.d]: ehm

**Comment by andralex** [comment]
noice

**Comment by andralex** [comment]
Auto-merge toggled on

**Pull request merged**